src/bin/pg_autoctl/primary_standby.c

```diff
@@ -766,8 +809,40 @@ standby_init_database(LocalPostgresServer *postgres,
 	}
 	else
 	{
-		if (!pg_basebackup(pgSetup->pgdata, pgSetup->pg_ctl, upstream))
+		/*
+		 * pg_basebackup has this bug where it will copy over the whole PGDATA
```

Comment:
is there a link to the bug mentioned here? Might be good to add to the comment as well

Reply:
No bug report / link that I know of, no, sorry. I intended to work on the fix myself, and then had a chat with folks on the Postgres IRC channel and gave the project away, but I didn't even track if it's been fixed or not yet. It's a very easy fix that's waiting to happen.

Basically `pg_basebackup` calls `waitpid` on the wal receiver child process at https://github.com/postgres/postgres/blob/master/src/bin/pg_basebackup/pg_basebackup.c#L2106 but that's when the file transfer is done already. So if the child process does an early exit because the replication slot does not exists, you see the error message at the top of the logs from `pg_basebackup`, but it still goes on with the base backup, and then fails after having copied a whole PGDATA that it doesn't even know how to re-use on the next attempt.